cargo-shuttle/src/lib.rs

```diff
@@ -251,20 +256,63 @@ impl Shuttle {
         })
     }
 
-    async fn check_lib_version(&self, project_args: ProjectArgs) -> Result<()> {
-        let cargo_path = project_args.working_directory.join("Cargo.toml");
-        let cargo_doc = read_to_string(cargo_path.clone())?.parse::<Document>()?;
-        let current_shuttle_version = &cargo_doc["dependencies"]["shuttle-service"]["version"];
-        let service_semver = match Version::parse(current_shuttle_version.as_str().unwrap()) {
+    fn check_lib_version(cargo_doc: Document, server_version: &Version) -> Result<()> {
+        fn get_version_from_table<'a>(table: &'a impl TableLike) -> Result<&'a Item> {
+            table
+                .get("version")
+                .ok_or_else(|| anyhow!("Missing 'version' key"))
+        }
+
+        fn get_item_as_str(item: &Item) -> Result<&str> {
+            item.as_str()
+                .ok_or_else(|| anyhow!("Expected string, found {}", item))
+        }
+
+        fn get_dependency_version_string<'a>(
```

Comment:
Ok, this is better. Any reason why the 2 levels of nesting is still needed?

Reply:
Purely stylistic 😁 . Can be better with code folding. I don't think the functions are that useful outside of the `check_lib_version` function so that is why they are in that scope. Can change if it looks complicated

Reply:
I see, this is fine :smile: